repos/spack_repo/builtin/packages/palace/package.py

```diff
@@ -97,68 +112,177 @@ class Palace(CMakePackage):
+            depends_on(f"slepc{cuda_variant}", when=f"+slepc{cuda_variant}")
+            depends_on(f"petsc{cuda_variant}", when=f"+slepc{cuda_variant}")
 
-    # More dependency variant conflicts
-    conflicts("^hypre+int64", msg="Palace uses HYPRE's mixedint option for 64 bit integers")
-    conflicts("^mumps+int64", msg="Palace requires MUMPS without 64 bit integers")
-    conflicts("^slepc+arpack", msg="Palace requires SLEPc without ARPACK")
+    # -- ROCm --
+    with when("+rocm"):
+        for arch in ROCmPackage.amdgpu_targets:
+            rocm_variant = f"+rocm amdgpu_target={arch}"
+            depends_on(f"hypre{rocm_variant}", when=f"{rocm_variant}")
+            depends_on(f"magma{rocm_variant}", when=f"{rocm_variant}")
+            depends_on(f"superlu-dist{rocm_variant}", when=f"+superlu-dist{rocm_variant}")
+            depends_on(f"libceed{rocm_variant}", when=f"{rocm_variant} @0.14:")
+            depends_on(f"strumpack{rocm_variant}", when=f"+strumpack{rocm_variant}")
+            depends_on(f"sundials{rocm_variant}", when=f"+sundials{rocm_variant} @0.14:")
+            depends_on(f"slepc{rocm_variant}", when=f"+slepc{rocm_variant}")
+            depends_on(f"petsc{rocm_variant}", when=f"+slepc{rocm_variant}")
 
     def cmake_args(self):
         args = [
             self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
             self.define_from_variant("PALACE_WITH_64BIT_INT", "int64"),
-            self.define_from_variant("PALACE_WITH_OPENMP", "openmp"),
-            self.define_from_variant("PALACE_WITH_SUPERLU", "superlu-dist"),
-            self.define_from_variant("PALACE_WITH_STRUMPACK", "strumpack"),
-            self.define_from_variant("PALACE_WITH_MUMPS", "mumps"),
-            self.define_from_variant("PALACE_WITH_SLEPC", "slepc"),
             self.define_from_variant("PALACE_WITH_ARPACK", "arpack"),
-            self.define_from_variant("PALACE_WITH_LIBXSMM", "libxsmm"),
-            self.define_from_variant("PALACE_WITH_MAGMA", "magma"),
+            self.define_from_variant("PALACE_WITH_CUDA", "cuda"),
             self.define_from_variant("PALACE_WITH_GSLIB", "gslib"),
+            self.define_from_variant("PALACE_WITH_HIP", "rocm"),
+            self.define_from_variant("PALACE_WITH_LIBXSMM", "libxsmm"),
+            self.define_from_variant("PALACE_WITH_MUMPS", "mumps"),
+            self.define_from_variant("PALACE_WITH_OPENMP", "openmp"),
+            self.define_from_variant("PALACE_WITH_SLEPC", "slepc"),
+            self.define_from_variant("PALACE_WITH_STRUMPACK", "strumpack"),
+            self.define_from_variant("PALACE_WITH_SUNDIALS", "sundials"),
+            self.define_from_variant("PALACE_WITH_SUPERLU", "superlu-dist"),
             self.define("PALACE_BUILD_EXTERNAL_DEPS", False),
         ]
 
+        # We guarantee that there are arch specs with conflicts above
+        if "+cuda" in self.spec:
```

Comment:
(nitpick) This is considered more idiomatic:
```suggestion
        if self.spec.satisfies("+cuda"):
```